### PR TITLE
Feature/gh login

### PR DESCRIPTION
**Pull request by kgajowy**

### CODE REVIEW

File: src/components/TopBar.js [hunk 1]
```diff
@@ -2,27 +2,31 @@ import React from 'react';
...
+        const buttonText = logged ? 'Wyloguj' : 'Zaloguj';
         return (
             <TopBarWrapper>
                 <Text type="basic" color={ Colors.white } display={ 'inline' }>
-                    Zaloguj się przez GitHub
+                    {!logged && 'Zaloguj się'}
+                    {logged && user && user.email}
                 </Text>
+                {pending && <ActivityIndicator/>}
+                {error && <p>Błąd: {JSON.stringify(error)}</p>}
```
Comment (Tuhaj) [inline src/components/TopBar.js]: Today we agreed on a meeting with @piecioshka that we're going to have an English version. I'm afraid it's good to start with i18n as soon as possible

File: src/shared/models/GithubUserModel.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { UserModel } from './UserModel';
+export class GithubUserModel extends UserModel {
+    static fromOAuth0(idTokenPayload) {
+        return Object.assign(new GithubUserModel(), idTokenPayload);
```
Comment (Tuhaj) [inline src/shared/models/GithubUserModel.js]: not sure, why is this here?

File: src/shared/services/OAuthProvider.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        let expiresAt = JSON.parse(localStorage.getItem('expires_at'));
+        return new Date().getTime() < expiresAt;
+    }
+    
+    static getCurrentUser() {
+        return GithubUserModel.fromOAuth0(JSON.parse(localStorage.getItem('user')));
+    }
+    
+    static logout() {
+        //this.auth0.logout(); //looking for a hint; it should be done to force auth0 to clear data but it reloads the page
```
Comment (Tuhaj) [inline src/shared/services/OAuthProvider.js]: maybe we should use `async`?

File: src/shared/models/GithubUserModel.js (hunk 2 above)
Comment (kgajowy) [reply src/shared/models/GithubUserModel.js]: @Tuhaj could you please explain more? for example, why do you think it should not be there?

It is just a DTO of user to rely on instead of a pure {} - a user will have a common representation as an specified object.  `UserModel` should have a base methods like `getDisplayName()` and `getEmail()`; if we are going to add different login methods, the DTO / fields names should does not matter (SOLID - Liskov substitution principle)

File: src/shared/services/OAuthProvider.js (hunk 3 above)
Comment (kgajowy) [reply src/shared/services/OAuthProvider.js]: Will check, however it will rather not help - `auth0.logout()` is reloading page by itself so it does not matter if I wait for it or not :(

**Comment by kgajowy** [comment]
@Tuhaj Please let me know if you still have any questions, as I would like to close this PR and continue in https://github.com/warsawjs/topics-manager/pull/20 (as this one has a history of commits which do not belong there - therefore I moved it to different branch)

**Review by Tuhaj — APPROVED** [review]
All good to me! ✅

**Comment by Tuhaj** [comment]
@kgajowy sorry! no go yet! We have a syntax error in the depending library

**Comment by Tuhaj** [comment]
screen shot 2018-06-06 at 18 08 45

**Comment by Tuhaj** [comment]
the login process works well, but this doesn't look nice ;-)
On master there was no error like that, can you check if you can also see it on your side? Moreover please run `npm install` again, cause I have a different package-lock.json
```
-        "string_decoder": {
-            "version": "1.1.1",
-            "resolved": "https://registry.npmjs.org/string_decoder/-/string_decoder-1.1.1.tgz",
-            "integrity": "sha512-n/ShnvDi6FHbbVfviro+WojiFzv+s8MPMHBczVePfUpDJLwoLT0ht1l4YwBCbi8pJAveEEdnkHyPyTP/mzRfwg==",
-            "requires": {
-                "safe-buffer": "5.1.2"
-            }
-        },
         "string-argv": {
             "version": "0.0.2",
             "resolved": "https://registry.npmjs.org/string-argv/-/string-argv-0.0.2.tgz",
@@ -12411,6 +12403,14 @@
                 }
             }
         },
+        "string_decoder": {
+            "version": "1.1.1",
+            "resolved": "https://registry.npmjs.org/string_decoder/-/string_decoder-1.1.1.tgz",
+            "integrity": "sha512-n/ShnvDi6FHbbVfviro+WojiFzv+s8MPMHBczVePfUpDJLwoLT0ht1l4YwBCbi8pJAveEEdnkHyPyTP/mzRfwg==",
+            "requires": {
+                "safe-buffer": "5.1.2"
+            }
+        },
         "stringify-object": {
             "version": "3.2.2",
             "resolved": "https://registry.npmjs.org/stringify-object/-/stringify-object-3.2.2.tgz",
```

**Comment by kgajowy** [comment]
Will continue in https://github.com/warsawjs/topics-manager/pull/20

File: public/index.html [hunk 4]
```diff
@@ -20,6 +20,7 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>React App</title>
+      <script type="text/javascript" src="../node_modules/auth0-js/build/auth0.js"></script>
```
Comment (piecioshka) [inline public/index.html]: wut 😄
